### PR TITLE
PSP-5154 : Increase test coverage for mayan service

### PR DESCRIPTION
**Pull request by eddherrera**
- Added unit tests for Document Service

### CODE REVIEW

File: source/backend/tests/unit/api/Services/DocumentServiceTest.cs [hunk 1]
```diff
@@ -15,17 +31,87 @@
...
+                new PimsDocumentTyp
+                {
+                    DocumentTypeId= 1,
+                    DocumentType="TYPE",
+                    MayanId=1,
+                }
+            });
+
+            // Act
+            var sut = service.GetPimsDocumentTypes();
```
Comment (devinleighsmith) [inline source/backend/tests/unit/api/Services/DocumentServiceTest.cs:83]: sut?

File: source/backend/tests/unit/api/Services/DocumentServiceTest.cs [hunk 2]
```diff
@@ -15,17 +31,87 @@
...
+            // Assert
+            sut.Should().Throw<NotAuthorizedException>();
+            documentRepository.Verify(x => x.Add(It.IsAny<PimsDocument>()), Times.Never);
+        }
 
-        #region Tests
-        #region UploadDocument
         [Fact]
-        public void UploadDocument_VirusScan()
+        public async void UploadDocumentAsync_UploadRequest_Sucess_NoMetadata()
```
Comment (devinleighsmith) [inline source/backend/tests/unit/api/Services/DocumentServiceTest.cs:114]: nit: success

File: source/backend/tests/unit/api/Services/DocumentServiceTest.cs [hunk 3]
```diff
@@ -34,22 +120,988 @@ public void UploadDocument_VirusScan()
...
+            Func<Task> sut =  async () => await service.GetStorageDocumentTypes(null, page: 1, pageSize: 10);
+
+            // Assert
+            sut.Should().Throw<NotAuthorizedException>();
+            documentStorageRepository.Verify(x => x.GetDocumentTypesAsync(null, It.IsAny<int>(), It.IsAny<int>()), Times.Never);
+
+        }
+
+        [Fact]
+        public async void GetStorageDocumenTypes_Success()
```
Comment (devinleighsmith) [inline source/backend/tests/unit/api/Services/DocumentServiceTest.cs:700]: nit: documenTypes -> documentTypes

File: source/backend/tests/unit/api/Services/DocumentServiceTest.cs [hunk 4]
```diff
@@ -34,22 +120,988 @@ public void UploadDocument_VirusScan()
...
+                .ReturnsAsync(new ExternalResult<QueryResult<DocumentType>>()
+                {
+                    HttpStatusCode = System.Net.HttpStatusCode.OK,
+                    Status = ExternalResultStatus.Success,
+                    Payload = new QueryResult<DocumentType>()
+                    {
+                        Count= 5,
+                        Results= new List<DocumentType>()
+                        {
+                            new() { Id = 1,},
```
Comment (devinleighsmith) [inline source/backend/tests/unit/api/Services/DocumentServiceTest.cs:721]: nit: 1,}

Comment (devinleighsmith) [reply source/backend/tests/unit/api/Services/DocumentServiceTest.cs:721]: is the comma after 1 required?

File: source/backend/tests/unit/api/Services/DocumentServiceTest.cs (hunk 1 above)
Comment (eddherrera) [reply source/backend/tests/unit/api/Services/DocumentServiceTest.cs:83]: subject under test

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**